In [10]:
library(arules)

In [9]:
library(Matrix)

In [15]:
groceries <- read.transactions("rexp/groceries.csv")
groceries # transaction과 item

Warning message in asMethod(object):
"removing duplicated items in transactions"


transactions in sparse format with
 9835 transactions (rows) and
 6871 items (columns)

In [16]:
itemFrequency(groceries[,1:3]) # 지지도 전체 트랜잭션 중 해당 아이템 거래비율

(appetizer) (appetizer),bathroom     (appetizer),cake 
        0.0026436197         0.0001016777         0.0001016777

install.packages("arules")
library(arules)
groceries<-read.transactions("groceries.csv", sep=",",,header = FALSE)
groceries
View(groceries)
summary(groceries)
#transactions as itemMatrix in sparse format with
#9835 rows (elements/itemsets/transactions) and
#169 columns (items) and a density of 0.02609146 
#9835행=거래건수, 169열=상품종류 개수, 0.026=밀도(행렬에서 0이 아닌 셀의 비율)
#sparse natrix = 희소행렬
#전체 행렬의 개수 * 밀도 
print(9835*169*0.026) # [1] 43214.99 = 0이 아닌 cell의 개수 
#한달동안 총 43000여개의 아이템이 판매되었다!
inspect() #세밀하게 점검하자 데이터가 잘 읽어졌는지 확인할 때 inspect사용
inspect(groceries[1:5]) #1번 부터 5번까지 항목들이 잘 읽어졌는지 확인 
groceries[,1:3] # 1째 열에서 3번째 열까지를 의미함. 
itemFrequency(groceries[, 1:3])
#지지도(support): 전체 거래 중 해당 아이템 거래 비율 지지도 

In [18]:
itemFrequencyPlot(groceries,support=0.1)

Warning message in min(height, na.rm = TRUE):
"min에 전달되는 인자들 중 누락이 있어 Inf를 반환합니다"
Warning message in max(graphics::strwidth(labels, units = "inches", cex = cex.names)):
"max에 전달되는 인자들 중 누락이 있어 -Inf를 반환합니다"


ERROR: Error in graphics::par(mai = mai): 그래픽 파라미터 "mai"에 지정된 값이 유효하지 않습니다
